modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/DefaultSourceCallback.java

```diff
@@ -308,7 +309,13 @@ public void onConnectionException(ConnectionException e) {
    */
   @Override
   public SourceCallbackContext createContext() {
-    return new DefaultSourceCallbackContext(this);
+    SourceCallbackContext context = new DefaultSourceCallbackContext(this);
+    try {
+      muleContext.getInjector().inject(context);
+    } catch (MuleException e) {
+      // ...
```

Comment:
Do I want not to fail in this case without a warning?
What will happen if there is a problem with the profiling service or other injected dependency. I think it would be better to fail now instead of in runtime with a NPE.